src/pointer-analysis/value_set.h

```diff
@@ -455,6 +455,16 @@ class value_sett
     exprt &expr,
     const namespacet &ns) const;
 
+  /// Get the entry for the symbol and suffix
+  /// \param identifier: The identifier for the symbol
+  /// \param expr_type: The type of the symbol (should not be a symbol itself)
```

Comment:
I don't think you should or have to make this restriction. Tag types should be ok, you just need to resolve them.